### PR TITLE
prelinking

### PR DESCRIPTION
**Pull request by philrhc**
refactors some of the prelinking logic from lib.rs into a new file prelink.rs
this starts an effort to make the architecture of the project more closely aligned with the three steps outlined in the presentation: prelink, link and init

https://docs.google.com/presentation/d/1kJ58_G1bApcRq0Ea1qipIVtC97GtMivoFFMzvTeeeqI/edit#slide=id.g27ac897f0bc_0_62

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -8,9 +8,10 @@ use {
     futures::future::FutureExt,
     heck::ToSnakeCase,
     indexmap::{IndexMap, IndexSet},
+    prelink::{embedded_helper_utils, embedded_python_standard_library},
```
Comment (dicej) [inline src/lib.rs:11]: Minor style nit: here you're importing these functions directly into scope, but later you call `prelink::search_for_libraries_and_configs` using the module qualifier (i.e. without importing it into scope).  It would be best to be consistent about that, and the prevailing style is to use the `module_name::function_name` style when calling a freestanding function defined in another module.

TL;DR: let's not import these directly and instead use `prelink::embedded_helper_utils` and `prelink::embedded_python_standard_library` to call those functions, respectively.

**Review by dicej — APPROVED** [review]
Looks great; thanks for doing this!

Just one minor style nit; see comment below.

File: src/lib.rs (hunk 1 above)
Comment (philrhc) [reply src/lib.rs]: fixed https://github.com/bytecodealliance/componentize-py/pull/107/commits/110147cf6dc52874624270160eb78cb7d3143e7a
